internal/impl/pure/scanner_json.go

```diff
@@ -0,0 +1,94 @@
+package pure
+
+import (
+	"context"
+	json "encoding/json"
+	"errors"
+	"io"
+
+	"github.com/benthosdev/benthos/v4/public/service"
+)
+
+const (
+	sjsonFieldContinueOnError = "continue_on_error"
+)
+
+func jsonDocumentScannerSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Stable().
+		Summary("Consumes JSON documents line-by-line.").
+		Fields(
+			service.NewBoolField(sjsonFieldContinueOnError).
+				Description("If a JSON decoding fails due to any error emit an empty message marked with the error and then continue consuming subsequent documents when possible.").
+				Default(false),
+		)
+}
+
+func init() {
+	err := service.RegisterBatchScannerCreator("json", jsonDocumentScannerSpec(),
```

Comment:
Let's maybe call this scanner `json_documents` to emphasise that it's a plurality of valid JSON documents rather than just one.